parser_tests/update_snapshots.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import sys
+from unittest import mock
+from urllib.request import urlopen
+
+from config import parsers
+from parser_tests.regression_test import get_canteen_url, \
+    get_snapshot_result_path, \
+    get_snapshot_website_path, \
+    parse_mocked, parsers_to_test
+
+base_directory = os.path.dirname(os.path.realpath(__file__))
+
+
+def generate_all_snapshots():
+    for parser, canteen in parsers_to_test:
+        generate_snapshot(parser, canteen)
+
+
+def generate_snapshot(parser, canteen):
+    intercepted_requests = {}
+
+    def intercept_request(requested_url):
+        response = urlopen(requested_url)
+
+        html = response.read()
+        intercepted_requests[requested_url] = html.decode('utf-8')
+
+        # cannot return original response, because read() has side effects on it
+        return urlopen(requested_url)
+
+    with mock.patch('urllib.request.urlopen', intercept_request):
+        url = get_canteen_url(parser, canteen)
+        parsers[parser].parse(url, canteen, 'full.xml')
+
+    # TODO: Improve mock so that parsers are free to use request library of their choice.
+    if len(intercepted_requests) == 0:
+        raise RuntimeError("Could not intercept the HTTP calls from parser '{}'. "
+                           "Please use `from urllib import request` "
+                           "and `request.urlopen`.".format(parser))
+
+    snapshot_website_path = get_snapshot_website_path(parser, canteen)
+    with open(snapshot_website_path, 'w', encoding='utf-8') as file:
+        json.dump(intercepted_requests, file, indent=4)
```

Comment:
Because a parser can call multiple websites, I found it easier to store them in one JSON file under their URL as a key. It was simply the easiest way for me to implement it. We could generate separate HTML-files.

I am not sure about the copyright. A website is meant to be transmitted and copied, right? 😐 And the point of these snapshots is just to cache a request, so that it can be reused later. You're making a good point, but I'm not seeing too much trouble here, although I'm not well educated in this area.

Reply:
> A website is meant to be transmitted and copied, right?

The problem is that we are redistributing the website with inclusion into the repo.

Reply:
I see the problem now. Do you know how to solve this issue? I'm not sure how to proceed.